dpnp/dpnp_iface_mathematical.py

```diff
@@ -3156,16 +3146,26 @@ def nan_to_num(x, copy=True, nan=0.0, posinf=None, neginf=None):
             )
         min_f = neginf
 
-    for part, part_out in zip(parts, parts_out):
-        nan_mask = dpnp.isnan(part)
-        posinf_mask = dpnp.isposinf(part)
-        neginf_mask = dpnp.isneginf(part)
+    if copy:
+        out = dpnp.empty_like(x)
+    else:
+        if not x.flags.writable:
+            raise ValueError("copy is required for read-only array `x`")
+        out = x
+
+    x_ary = dpnp.get_usm_ndarray(x)
+    out_ary = dpnp.get_usm_ndarray(out)
+
+    q = x.sycl_queue
+    _manager = dpu.SequentialOrderManager[q]
+
+    h_ev, comp_ev = ufi._nan_to_num(
+        x_ary, nan, max_f, min_f, out_ary, q, depends=_manager.submitted_events
+    )
 
-        part = dpnp.where(nan_mask, nan, part, out=part_out)
-        part = dpnp.where(posinf_mask, max_f, part, out=part_out)
-        part = dpnp.where(neginf_mask, min_f, part, out=part_out)
+    _manager.add_event_pair(h_ev, comp_ev)
 
-    return out
+    return dpnp.get_result_array(out) if copy else x
```

Comment:
It seems the same as
```suggestion
    return out
```
Or why do we need to call `dpnp.get_result_array(out)` here?

Reply:
The `if copy else x` part replicated the old behavior of returning the original `usm_ndarray` when `copy` is `False`, but I think this was a mistake in retrospect. The `get_result_array` is still necessary for this case, or else, `out` will be a `usm_ndarray`

Reply:
Oh, now I got it, Thank you.